code/modules/mob_spawn/ghost_roles/golem_roles.dm

```diff
@@ -71,6 +69,14 @@
 	else
 		new_spawn.mind.set_assigned_role(SSjob.GetJobType(/datum/job/free_golem))
 
+/obj/effect/mob_spawn/ghost_role/human/golem/proc/try_keep_home(mob/new_spawn)
+	var/static/list/allowed_areas
+	if (!allowed_areas)
+		allowed_areas = typecacheof(list(/area/icemoon, /area/lavaland, /area/ruin))
```

Comment:
you dont need to check for this it will assign the static var to the output of the proc once ever